client/reader/tag-stream/main.jsx

```diff
@@ -20,28 +19,45 @@ import smartSetState from 'lib/react-smart-set-state';
 import * as stats from 'reader/stats';
 import HeaderBack from 'reader/header-back';
 
+
 const TagStream = React.createClass( {
 
+	_isMounted: false,
+
 	propTypes: {
 		tag: React.PropTypes.string
 	},
 
 	smartSetState: smartSetState,
 
 	getInitialState() {
+		const title = this.getTitle();
 		return {
-			title: this.getTitle(),
+			title,
 			subscribed: this.isSubscribed(),
-			canFollow: has( ReaderTags.get( this.props.tag ), 'ID' )
+			canFollow: has( ReaderTags.get( this.props.tag ), 'ID' ),
+			isEmojiTitle: title && twemoji.test( title )
 		};
 	},
 
+	componentWillMount() {
+		var self = this;
+		this._isMounted = true;
+		// can't use arrows with asyncRequire
+		asyncRequire( 'emoji-text', function( emojiText ) {
+			if ( self._isMounted ) {
```

Comment:
https://facebook.github.io/react/blog/2015/12/16/ismounted-antipattern.html

but I have a feeling this is one of the exceptions because i can't fathom how to cancel this callback in an unmount